core/src/datayoga_core/blocks/add_field/tests/test_add_field.py

```diff
@@ -21,7 +22,7 @@ async def test_add_nested_field():
                   "expression": "[name.fname,name.lname] | join(' ', @)"})
     block.init()
     assert await block.run([{"name": {"fname": "john", "lname": "doe"}}]) == ([
-        {"name": {"fname": "john", "lname": "doe", "full_name": "john doe"}}], [Result.SUCCESS])
```

Comment:
why the change from the Enum?

Reply:
It's not Enum anymore, it's a Result object with SUCCESS status.

Reply:
Maybe use @dataclass unless we actually have business logic in those? This mode of a 'factory' singleton just seems unnatural. especially since we are creating one as a constant, but then duplicating it.

Reply:
From @zalmane :
Here is my suggestion. I would not mix static and non static methods in that dataclass. It is really meant as a Struct or NamedDict:
result.py
success= Result(Status.SUCCESS)
@dataclass
class Result():
    status: Status
    message: Optional[str] = None
  
    @staticmethod - remove this altogether. the code can just do this one liner and makes it more readable and easier to maintain
    def reject(message: Optional[str] = None) -> Result:
        return Result(Status.REJECTED, message)
so you use it:
import result
result.success

Reply:
done